src/storage/src/hummock/sstable_store.rs

```diff
@@ -918,6 +923,59 @@ impl StreamingSstableWriterFactory {
     }
 }
 
+pub enum UnifiedSstableWriterFactor {
```

Comment:
Given that we already introduced `UnifiedSstableWriter`, we don't need to introduce a enum for `UnifiedSstableWriterFactory`. We can just have:
```rust
pub struct UnifiedSstableWriterFactory {
    sstable_store: SstableStoreRef,
}

#[async_trait::async_trait]
impl SstableWriterFactory for UnifiedSstableWriterFactory {
    type Writer = UnifiedSstableWriter;

    async fn create_sst_writer(
        &mut self,
        object_id: HummockSstableObjectId,
        options: SstableWriterOptions,
    ) -> HummockResult<Self::Writer> {
        if self.sstable_store.store().support_streaming_upload() {
            ...
            Ok(UnifiedSstableWriter::StreamingSstableWriter(
                streaming_uploader_writer,
            ))
        } else {
            ...
            Ok(UnifiedSstableWriter::BatchSstableWriter(
                batch_uploader_writer,
            ))
        }
    }
}
```

In this case, compactor doesn't need to check `support_streaming_upload` explicitly

Reply:
Indeed, changed. Next pr will try to refactor [this logic](https://github.com/risingwavelabs/risingwave/blob/69604f2873b37714495b954fead0f8a9cc78a952/src/storage/src/hummock/compactor/mod.rs#L150) as well.